plugins/Facebook/class.facebook.plugin.php

```diff
@@ -552,7 +553,7 @@ public function authorizeUri($query = false, $redirectUri = false) {
             'client_id' => $appID,
             'redirect_uri' => $redirectUri,
             'scope' => $scopes,
-            'state' => json_encode(['token' => $stateToken]),
+            'state' => json_encode(['token' => $stateToken, 'Target' => $this->getTargetUri()]),
```

Comment:
Let's use camel case field names like token does.
You can do `$form->setFormValue('Target', val('target', $state, '/'));` above.